tests/foreman/cli/test_computeresource_azurerm.py

```diff
@@ -0,0 +1,629 @@
+        Provisions the host on AzureRM using UserData template
+        Later in tests this host will be used to perform assertions
+        """
+        hammer_api_timeout_set()
+        skip_yum_update_during_provisioning(template='Kickstart default user data')
+        host = Host.create(
+            {
+                'name': self.hostname,
+                'compute-attributes': self.compute_attrs,
+                'interface': self.interfaces_attributes,
+                'image': module_azurerm_cloudimg.name,
+                'hostgroup': azurerm_hostgroup.name,
+                'location': module_location.name,
+                'organization': module_org.name,
+                'operatingsystem-id': module_os.id,
+            },
+            timeout=1800,
+        )
+        yield host
+        skip_yum_update_during_provisioning(template='Kickstart default user data', reverse=True)
+        Host.delete({'name': self.fullhostname}, timeout=1800)
+        hammer_api_timeout_set(reverse=True)
+
+    @pytest.fixture(scope='class')
+    def azureclient_host(self, azurermclient, class_host_ud):
+        """Returns the AzureRM Client Host object to perform the assertions"""
+        return azurermclient.get_vm(name=class_host_ud['name'].split('.')[0])
+
+    @upgrade
+    @tier3
+    def test_positive_azurerm_host_provisioned(
+        self,
+        class_host_ud,
+        azureclient_host,
+        module_azurerm_cr,
+        module_azurerm_cloudimg,
+        azurerm_hostgroup,
+    ):
+        """Host can be provisioned on AzureRM
+
+        :id: c99d2679-1742-4ef3-9288-2961d18a30e7
+
+        :CaseLevel: Component
+
+        ::CaseImportance: Critical
+
+        :steps:
+            1. Create a AzureRM Compute Resource and provision host.
+
+        :expectedresults:
+            1. The host with hostgroup should be provisioned on AzureRM
+            2. The host name should be the same as given in data to provision the host
+            3. The host should show "Pending installation" status for provisioned host(
+               as Satellite and VM are in diff networks hence status shows as "Pending
+               installation")
+            4. The provisioned host should be assigned with external IP
+            5. The host Name and Platform should be same on Azure Cloud as provided during
+               provisioned.
+        """
+        assert class_host_ud['name'] == self.fullhostname
+        assert class_host_ud['status']['build-status'] == "Pending installation"
+        assert class_host_ud['network-interfaces'][0]['ipv4-address'] == azureclient_host.ip
+        assert class_host_ud['compute-resource'] == module_azurerm_cr.name
+        assert class_host_ud['operating-system']['image'] == module_azurerm_cloudimg.name
+        assert class_host_ud['host-group'] == azurerm_hostgroup.name
+
+        # Azure cloud
+        assert self.hostname.lower() == azureclient_host.name
+        assert self.vm_size == azureclient_host.type
```

Comment:
Is it really needed to have two separate classes for provisioning? It's a bit hard to wrap my head around this but it seems there's just a difference in the image (which is not even asserted later).

Reply:
yes it's needed as this PR is part-1 to add support around UD and FT provisioning ... so there would be few more tests where they either requires put under UD or FT class ...

hence for easy of separation (more utilization) this requires ..